pkg/platform/provider/baremetal/manifests/cilium/eni-ipamd.yaml

```diff
@@ -0,0 +1,233 @@
+---
+# ipamd configmap
+apiVersion: v1
+data:
+  TKE_ENI_IPAMD_VPC_ID: vpc-k4l6pld3
+kind: ConfigMap
+metadata:
+  name: tke-eni-ipamd
+  namespace: kube-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: tke-eni-ipamd
+  namespace: kube-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: tke-eni-ipamd
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+      - namespaces
+      - nodes
+      - nodes/status
+    verbs:
+      - list
+      - watch
+      - get
+      - patch
+      - update
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+      - endpoints
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+      - update
+      - create
+      - patch
+  - apiGroups:
+      - apps
+      - extensions
+    resources:
+      - statefulsets
+      - deployments
+    verbs:
+      - list
+      - watch
+  - apiGroups:
+      - apiextensions.k8s.io
+    resources:
+      - customresourcedefinitions
+    verbs:
+      - create
+      - update
+      - get
+  - apiGroups:
+      - networking.tke.cloud.tencent.com
+    resources:
+      - staticipconfigs
+      - underlayips
+      - nodeeniconfigs
+      - vpcipclaims
+      - vpcips
+      - vpcenis
+      - eipclaims
+    verbs:
+      - create
+      - update
+      - delete
+      - get
+      - list
+      - watch
+      - patch
+  - apiGroups:
+      - cilium.io
+    resources:
+      - ciliumnodes
+      - ciliumnodes/status
+      - ciliumnodes/finalizers
+    verbs:
+      - create
+      - update
+      - delete
+      - get
+      - list
+      - watch
+      - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: tke-eni-ipamd
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: tke-eni-ipamd
+subjects:
+  - kind: ServiceAccount
+    name: tke-eni-ipamd
+    namespace: kube-system
+---
+# ipamd deployment
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    k8s-app: tke-eni-ipamd
+    qcloud-app: tke-eni-ipamd
+  name: tke-eni-ipamd
+  namespace: kube-system
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      k8s-app: tke-eni-ipamd
+      qcloud-app: tke-eni-ipamd
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        k8s-app: tke-eni-ipamd
+        qcloud-app: tke-eni-ipamd
+    spec:
+      affinity:
+        nodeAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - preference:
+                matchExpressions:
+                  - key: node-role.kubernetes.io/master
```

Comment:
should handle 1.20+ k8s, maybe just has `node-role.kubernetes.io/control-plane`.

Reply:
add 
        - matchExpressions:
              - key: node-role.kubernetes.io/control-plane
                operator: Exists

Reply:
ref https://github.com/tkestack/tke/pull/1293/files